ctx.go

```diff
@@ -370,10 +371,60 @@ func (c *DefaultCtx) Response() *fasthttp.Response {
 	return &c.fasthttp.Response
 }
 
+type Fmt struct {
+	MediaType string
+	Handler   func(Ctx) error
+}
+
 // Format performs content-negotiation on the Accept HTTP header.
+// It uses Accepts to select a proper format and calls the matching
+// user-provided handler function.
+// If no accepted format is found, and a format with MediaType "default" is given,
+// that default handler is called. If no format is found and no default is given,
+// StatusNotAcceptable is sent.
+func (c *DefaultCtx) Format(handlers ...Fmt) error {
+	if len(handlers) == 0 {
+		panic("Format requires at least one handler")
+	}
+
+	if c.Get(HeaderAccept) == "" {
+		c.Response().Header.SetContentType(handlers[0].MediaType)
+		return handlers[0].Handler(c)
+	}
+
+	types := make([]string, 0, 8)
```

Comment:
Why was capacity set as 8?

Reply:
errors.New should be used right? @efectn

Reply:
My guess to avoid runtime allocs if it's under 8, which is probably a reasonable limit

Reply:
> My guess to avoid runtime allocs if it's under 8, which is probably a reasonable limit

Yeah probably. It would be really nice if he adds a comment.

Reply:
using a static size allows stack allocation, and 8 is an approximation of how many content types a user might use. The number was selected completely arbitrarily.  Even though it's stack allocated, making the slice bigger makes it slower.  I'll add this info as a comment as suggested